### PR TITLE
Add Test plan for Quarkus-959

### PR DESCRIPTION
**Pull request by pjgg**
Dev Service simplifies testing and Quarkus dev-mode with containers.

As a developer, you spend too much time in unproductive cycles such us:

```
Write Code → Compile → Deploy → Refresh Browser → Repeat
```

But Quarkus allow you to pick up your changes on the fly thanks to Quarkus dev-mode, and also allows you to deploy some developer services such as BDs or Kafka with almost zero configuration thanks to dev-services. 

 Quarkus Dev-mode developer cycle:

```
Write Code → Refresh Browser → Repeat
```
Note: all your service DBs will be deployed and refresh it on the fly, allowing you to develop your services following a continuous testing approach. 

This PR is a test plan to cover the supported features scoped for RHBQ 2.2 related to Dev-services

### CODE REVIEW

**Comment by rsvoboda** [comment]
Please update the PR to comply with https://gitlab.cee.redhat.com/quarkus-qe/signpost/-/blob/main/pull-requests.md#main-rule-set-for-pull-requests

**Review by rsvoboda — CHANGES_REQUESTED** [review]
Also miss some details like:

 - Extend CRUD test cases with applications that spin up the database dev services
 - Extend messaging test cases with applications that spin up messaging dev services
 - Use the existing dev mode test coverage to ensure that running scenarios without dev services is still possible 
 - Cover DEV UI
 - Quarkus app is waiting for the necessary services to start, is developer informed about started services progress?

File: QUARKUS-959.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Quarkus-959 - Improved Developer Joy with DevServices
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-959
+
+The goal is to verify proper functionality of the Quarkus DevServices
+
+## Scope of the testing
+Test development will focus on the supported features scoped for RHBQ 2.2:
+ - All persisted databases which we have extensions (Postgresql, MariaDB, Mysql, Oracle, DB2)
```
[the remark this answers is not in the document]
Comment (pjgg) [reply QUARKUS-959.md]: Yes, I've removed both databases from the test plan. We could extend it in the future once will be supported.

File: QUARKUS-959.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ - Some messaging extensions such as AMQP, Strimzi Kafka, RedPanda Kafka
```
[the remark this answers is not in the document]
Comment (pjgg) [reply QUARKUS-959.md]: Strimzi Kafka looks that is not ready, so lets remove it from now and we could add it in the future.

File: QUARKUS-959.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## Scope of the testing
+Test development will focus on the supported features scoped for RHBQ 2.2:
+ - All persisted databases which we have extensions (Postgresql, MariaDB, Mysql, Oracle, DB2)
+ - Some messaging extensions such as AMQP, Strimzi Kafka, RedPanda Kafka 
+
+### Impact on testsuites and testing automation:
+ - Enhance Quarkus QE test framework to cover all the features that are part of the scope
+ - Add test coverage into the Quarkus QE test suite only for baremetal
+ - Ensure this coverage works both in JVM and NATIVE mode
```
[the remark this answers is not in the document]
Comment (pjgg) [reply QUARKUS-959.md:25]: I've covered this subject with the following point:

```
A developer must be able to run test with dev-services enabled and also run the same service in dev mode. Both services (test + dev-services and quarkus dev-mode + dev-services) must be not interfere with each other.
```

Let me know If I am missing something else, please.

File: QUARKUS-959.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+## Scope of the testing
+Test development will focus on the supported features scoped for RHBQ 2.2:
+ - All persisted databases which we have extensions (Postgresql, MariaDB, Mysql, Oracle, DB2)
+ - Some messaging extensions such as AMQP, Strimzi Kafka, RedPanda Kafka 
+
+### Impact on testsuites and testing automation:
+ - Enhance Quarkus QE test framework to cover all the features that are part of the scope
+ - Add test coverage into the Quarkus QE test suite only for baremetal
+ - Ensure this coverage works both in JVM and NATIVE mode 
+ - Make sure the service is started just once and not respinned multiple times
```
[the remark this answers is not in the document]
Comment (pjgg) [reply QUARKUS-959.md:26]: covered by the following point:

```
Developers must be able to develop his services following a continuos testing approach 
```

**Comment by pjgg** [comment]
> I have one final item to cover.
> 
> What if user has defined test resources to start the database. What will happen if the tests are executed, will it collide with DevServices / is there some automagic to detect this?

Well, in the exploratory test that we did, the test is pointing to a SQL init script that is executed at the begging by  hibernate ORM
Properties -> https://github.com/quarkus-qe/quarkus-test-suite/blob/main/sql-db/sql-app/src/test/resources/postgresql.properties#L2

Yes, but is true we should add to our test plan a point that cover "DB evolution" quarkus extensions as Flyway or Liquidbase(TP)

**Review by rsvoboda — APPROVED** [review]
I think it's OK to proceed, the TP can be evolved if we spot new area for coverage as we go.

**Pull request merged**